src/Xamarin.Android.Build.Tasks/Tasks/CheckClientHandlerType.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using Microsoft.Android.Build.Tasks;
+using Java.Interop.Tools.Cecil;
+using Mono.Cecil;
+
+namespace Xamarin.Android.Tasks
+{
+	public class CheckClientHandlerType : AndroidTask
+	{
+		public override string TaskPrefix => "CCHT";
+
+		[Required]
+		public string ClientHandlerType { get; set; }
+		[Required]
+		public string ValidHandlerType { get; set; }
+		[Required]
+		public ITaskItem[] ResolvedAssemblies { get; set; }
+
+		public bool IsLinkerEnabled { get; set; }
+
+		public override bool RunTask ()
+		{
+			string[] types = ClientHandlerType.Split (',');
+			string type = types[0].Trim ();
+			string assembly = "Mono.Android";
+
+			if (types.Length > 1) {
+				assembly = types[1].Trim ();
+			}
+			// load the assembly.
+			ITaskItem foundAssembly = null;
+			foreach (var asm in ResolvedAssemblies) {
+				string filename = Path.GetFileNameWithoutExtension (asm.ItemSpec);
+				if (string.CompareOrdinal (assembly, filename) == 0) {
+					foundAssembly = asm;
+					break;
+				}
+			}
+			if (foundAssembly == null) {
+				Log.LogCodedError ("XA1033", Xamarin.Android.Tasks.Properties.Resources.XA1033, assembly);
+				return !Log.HasLoggedErrors;
+			}
+			// find the type.
+			var readerParameters = new ReaderParameters {
+				ReadSymbols = false,
+			};
+			using (var resolver = new DirectoryAssemblyResolver (this.CreateTaskLogger (), loadDebugSymbols: false, loadReaderParameters: readerParameters)) {
+				foreach (var asm in ResolvedAssemblies) {
+					var path = Path.GetFullPath (Path.GetDirectoryName (asm.ItemSpec));
+					if (!resolver.SearchDirectories.Contains (path)) {
+						resolver.SearchDirectories.Add (path);
+					}
+				}
+
+				var assemblyDefinition = resolver.GetAssembly (Path.GetFullPath (foundAssembly.ItemSpec));
+				TypeDefinition handlerType = null;
+				foreach (var model in assemblyDefinition.Modules) {
+					handlerType = assemblyDefinition.MainModule.GetType (type);
+					if (handlerType != null)
+						break;
+				}
+				if (handlerType == null) {
+					if (IsLinkerEnabled) {
```

Comment:
Why would this be "fine"?  If the type can't be found, things are gonna be weird, even if the linker is enabled.  Why only log a Debug message and return *true*?

Reply:
I'm gonna have to rework this a bit. 

The problem this was trying to solve was in Release mode if a `HttpClient` is NOT in the code, the entire handler type is removed by the link. So building in Release mode in this instance would give you a build error. 

If the user IS using a `HttpClient` then this does not happen as the type is preserved. 

This does not happen in Debug modes, so you get the error only if the type is wrong. 

I think I'll need to run this task earlier in the process... before the linker in order to get consistent behaviour. I'll test that now.

Reply:
Looks like this was invalidated, as `IsLinkerEnabled` no longer exists.  Not sure why GitHub doesn't show this comment as being obsolete.

Reply:
Doubly weird, clicking "Resolve conversation" does nothing. :-/